### PR TITLE
Update vendored zstd to 1.4.0

### PR DESCRIPTION
**Pull request by bai**
Update vendored zstd from 1.3.8 to 1.4.0.

Quick bench (macOS):
```
$ benchstat old.txt new.txt
name                    old time/op   new time/op   delta
StreamCompression-12     91.8ms ± 2%   91.8ms ± 1%    ~     (p=1.000 n=5+4)
StreamDecompression-12   15.9ms ± 2%   16.5ms ± 2%  +3.40%  (p=0.008 n=5+5)
Compression-12           90.8ms ± 1%   92.2ms ± 3%    ~     (p=0.222 n=5+5)
Decompression-12         13.5ms ± 2%   13.8ms ± 1%    ~     (p=0.056 n=5+5)

name                    old speed     new speed     delta
StreamCompression-12    109MB/s ± 2%  109MB/s ± 1%    ~     (p=1.000 n=5+4)
StreamDecompression-12  626MB/s ± 2%  605MB/s ± 2%  -3.29%  (p=0.008 n=5+5)
Compression-12          110MB/s ± 1%  108MB/s ± 3%    ~     (p=0.222 n=5+5)
Decompression-12        741MB/s ± 2%  725MB/s ± 1%    ~     (p=0.056 n=5+5)
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 [C Zstd Homepage](https://github.com/Cyan4973/zstd)
 
-The current headers and C files are from *v1.3.8* (Commit
-[470344d](https://github.com/facebook/zstd/releases/tag/v1.3.8)).
+The current headers and C files are from *v1.4.0* (Commit
+[470344d](https://github.com/facebook/zstd/releases/tag/v1.4.0)).
```
Comment (Viq111) [inline README.md]: nit: mind changing the commit hash ?

**Review by Viq111 — COMMENTED** [review]
Thanks for pulling so fast!

Changes looks good to me (just added a tiny comment)
I'll wait ~a day to see if any issues come out at https://github.com/facebook/zstd/issues but otherwise will merge tomorrow.

One nice feature we should change once it's stable is https://github.com/facebook/zstd/releases/tag/v1.4.0
> A new experimental function [ZSTD_decompressBound()](https://github.com/facebook/zstd/blob/a880ca239b447968493dd2fed3850e766d6305cc/lib/zstd.h#L1178)

This would help us get rid of the deprecated `ZSTD_getDecompressedSize `: https://github.com/DataDog/zstd/blob/8e0d46c/zstd.go#L123

File: README.md (hunk 1 above)
Comment (bai) [reply README.md]: Good catch, fixed 👍

**Review by Viq111 — APPROVED** [review]

**Comment by Viq111** [comment]
Thanks again for your contribution!

**Pull request merged**